### DISCUSSION TITLE
Need help on building shared libraries on Windows machine for Android x86_64 (emulator)

### DISCUSSION THREAD

**Discussion by cmpktheo** [ggml-org/llama.cpp#7394] [Q&A]
Hello,
I am using a windows development machine and need a consistent way to build llama.cpp for the Android x86_64 emulator (not aarch64).
I have tried to build the shared libraries using termux on the x86_64 emulator. Everything is build except the shared libraries (so files).
If anyone has attempted it, please advise.
Thanks.

Edit:
I used Android Studio + NDK to build for android x86_64.
x86_64-linux-android33-clang -shared -o libggml-shared.so -fPIC ggml.c ggml-alloc.c ggml-backend.c ggml-quants.c -Wall

x86_64-linux-android33-clang++.cmd -shared -o libllama.so -fPIC llama.cpp -Wall -L. -lggml-shared

Then tried to load the resulting libllama.so in a test MAUI app with no success.
Can anyone pinpoint if I am missing something?

**Comment by sunzhe09** [comment]
+1
